sources/packages/backend/apps/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -321,15 +321,23 @@ export class ApplicationStudentsController extends BaseController {
     const student = await this.studentService.getStudentById(
       studentToken.studentId,
     );
+    // If offering is present, the selected offering's start and end dates will be used.
+    let referenceStudyStartDate = payload.data.selectedOfferingDate;
+    let referencedStudyEndDate = payload.data.selectedOfferingEndDate;
+    if (!payload.data.selectedOffering) {
+      // If offering is not present, the PIR provided study start and end dates will be used instead.
+      referenceStudyStartDate = payload.data.studyStartDate;
+      referencedStudyEndDate = payload.data.studyEndDate;
```

Comment:
Please use the dry run submission result values to perform validations and refrain from using the payload.data directly. I know the existing code was using the payload data.